### ISSUE TITLE
Failed to replace PagedAttentionWithRoPE.

### ISSUE THREAD

**Issue by hotdog-zz** [vllm-project/vllm#682]
In my test.py, I intend to test and enhance the model's performance with longer input sequences. To achieve this, I need to modify the ModelConfig and PagedAttentionWithRoPE. However, I've encountered difficulties only when trying to modify the PagedAttentionWithRoPE component. More specifically:

    from vllm import LLM, SamplingParams
    from vllm.model_executor.layers.attention import PagedAttentionWithRoPE, PagedAttention
    from vllm.config import ModelConfig, _get_and_verify_dtype
    import torch
    from typing import List, Optional
    from vllm.transformers_utils.config import get_config
    import pdb

    ratio = 4
    def __Configinit__(
            self,
            model: str,
            tokenizer: str,
            tokenizer_mode: str,
            trust_remote_code: bool,
            download_dir: Optional[str],
            use_np_weights: bool,
            use_dummy_weights: bool,
            dtype: str,
            seed: int,
        ) -> None:
            pdb.set_trace()
        
            self.model = model
            self.tokenizer = tokenizer
            self.tokenizer_mode = tokenizer_mode
            self.trust_remote_code = trust_remote_code
            self.download_dir = download_dir
            self.use_np_weights = use_np_weights
            self.use_dummy_weights = use_dummy_weights
            self.seed = seed

            self.hf_config = get_config(model, trust_remote_code)
            self.dtype = _get_and_verify_dtype(self.hf_config, dtype)
            self._verify_tokenizer_mode()
    ModelConfig.__init__ = __Configinit__

    def __Ropeinit__(
            self,
            num_heads: int,
            head_size: int,
            scale: float,
            rotary_dim: int,
            max_position: int = 8192,
            base: int = 10000,
            num_kv_heads: Optional[int] = None,
        ) -> None:
            super(PagedAttention, self).__init__(num_heads, head_size, scale, num_kv_heads)
        
            pdb.set_trace()
        
            # Create the cos and sin cache.
            inv_freq = 1.0 / (base**(torch.arange(0, rotary_dim, 2) / rotary_dim))
            t = torch.arange(max_position).float()
            freqs = torch.einsum("i,j -> ij", t, inv_freq.float())
            cos = freqs.cos()
            sin = freqs.sin()
            cache = torch.cat((cos, sin), dim=-1)
            # FIXME(woosuk): This assumes that we configure the default dtype when
            # initializing the model.
            # TODO(woosuk): Make it more robust.
            torch_dtype = torch.get_default_dtype()
            cache = cache.to(torch_dtype)
            # Embedding size: [max_position, rotary_dim]
            self.register_buffer("cos_sin_cache", cache, persistent=False)
    PagedAttentionWithRoPE.__init__ = __Ropeinit__

    llm = LLM(model="OpenBuddy/openbuddy-openllama-7b-v5-fp16",tensor_parallel_size=4,        
    max_num_batched_tokens=2048*ratio)
    sample = SamplingParams(
        max_tokens=128
    )
    output = llm.generate(prompt + "\nAssistant:", sampling_params=sample)

After executing the aforementioned code, it successfully reaches the breakpoint I set within the __Configinit__ function, but it doesn't enter the breakpoint within the __Ropeinit__ function. Instead, it continues to enter the PagedAttentionWithRoPE.__init__ function. I'm unsure about how I should proceed with the necessary modifications.

**Comment by Zhuqln** [comment]
they are working on this PR
https://github.com/vllm-project/vllm/issues/674#issuecomment-1666383693

**Comment by hotdog-zz** [comment]
> they are working on this PR [#674 (comment)](https://github.com/vllm-project/vllm/issues/674#issuecomment-1666383693)

Thanks.
